File: elixir/beer-song/lib/beer_song.ex
defmodule BeerSong do
  def verse(n), do: "#{current(n)}\n#{next(n)}\n"
  def lyrics(range \\ 99..0), do: range |> Enum.map(&verse/1) |> Enum.join("\n")

  defp current(n), do: "#{bottles(n) |> String.capitalize()} of beer on the wall, #{bottles(n)} of beer."
  defp next(n), do: "#{action(n)}, #{bottles(Integer.mod(n - 1, 100))} of beer on the wall."

  defp bottles(0), do: "no more bottles"
  defp bottles(1), do: "1 bottle"
  defp bottles(n), do: "#{n} bottles"

  defp action(0), do: "Go to the store and buy some more"
  defp action(n), do: "Take #{subject(n)} down and pass it around"

  defp subject(1), do: "it"
  defp subject(_), do: "one"
end

File: csharp/beer-song/BeerSong.cs
using System.Linq;

public static class BeerSong
{
    private static string Verse(int number) =>
        number switch
        {
            0 => "No more bottles of beer on the wall, no more bottles of beer.\nGo to the store and buy some more, 99 bottles of beer on the wall.",
            1 => "1 bottle of beer on the wall, 1 bottle of beer.\nTake it down and pass it around, no more bottles of beer on the wall.",
            2 => "2 bottles of beer on the wall, 2 bottles of beer.\nTake one down and pass it around, 1 bottle of beer on the wall.",
            _ => $"{number} bottles of beer on the wall, {number} bottles of beer.\nTake one down and pass it around, {number - 1} bottles of beer on the wall."
        };

    public static string Recite(int startBottles, int takeDown) =>
        string.Join("\n\n", Enumerable.Range(startBottles - takeDown + 1, takeDown).Reverse().Select(Verse));
}